pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -403,7 +396,11 @@ def barrier(self, *args, **kwargs) -> None:
             torch.distributed.barrier()
 
     def broadcast(self, obj: object, src: int = 0) -> object:
-        return self.dist.broadcast(obj)
+        obj = [obj]
+        if self.global_rank != 0:
+            obj = [None] * len(obj)
```

Comment:
this is not correct: obj is not guaranteed to be a list. and this will cause surprised with strings

Reply:
it been converted to a string at line 399

obj = [obj]

Reply:
it's possible we can save an extra list serialization by checking if `isinstance(obj, list)` and calling `broadcast_object_list` directly in that case w/o needing to convert it

Reply:
Current logic seems works really wired with list obj. I'm not sure whether check list before convert is a breaking change or not
```
if obj = [1,2,3]
obj= [obj] -> obj = [[1,2,3]]
```

Reply:
why would it be a breaking change? since we can post-process the return of `broadcast_object_list` before we return, there should be no leakage to the caller. its a minor efficiency point is all. but we can discuss this further with better support for broadcasting tensors in a separate PR

Reply:
nit: won't len(obj) always be 1?

Reply:
It's not nit...I feel we can revisit this part of the logic in a separate issue/PR. Do we need set a None list on the ranks broadcast too?

Reply:
> Do we need set a None list on the ranks broadcast too?

yes, each rank must provide lists of equal sizes: https://pytorch.org/docs/stable/_modules/torch/distributed/distributed_c10d.html#broadcast_object_list

but we don't need to multiply by len(obj) here.

Reply:
```suggestion
            obj = [None]
```